### ISSUE TITLE
aws_emr_instance_group can't be destroyed

### ISSUE THREAD

**Issue by synhershko** [terraform-providers/terraform-provider-aws#1355]
Given a definition like below:

```
resource "aws_emr_instance_group" "querying-task-nodes" {
  name                      = "task"
  cluster_id                = "${aws_emr_cluster.querying.id}"
  instance_type             = "m4.xlarge"
  instance_count            = 5
}
```

If the EMR cluster is being destroyed successfully the instance group is getting into a limbo and can't be destroyed via terraform, giving the following error:

```
* aws_emr_instance_group.querying-task-nodes: ValidationException: An instance group may only be modified when the cluster is running or waiting.
	status code: 400, request id: d9555f4d-791e-11e7-b51e-cf8d28f27320
```

From what I can gather the issue is with the IG not being removed before the cluster is destroyed.

Using latest terraform (Terraform v0.10.0)

**Comment by timothyclifford** [comment]
Any updates? 

We are seeing same behaviour with `v0.11.2`

**Comment by sbraden** [comment]
I recently observed this behavior on AWS. The work around (through the AWS Cloud Formation console) is when deleting the stack, opt to retain the resource (in this case the aws_emr_instance_group) that prevented the successful deletion of the stack. Then the delete of the stack can continue successfully.

**Comment by utsav2307** [comment]
if the cluster with that task group is deleted, you can use terraform state rm to delete that entry from terraform state, and rerun your terraform to create cluster.

**Comment by bflad** [comment]
The fix for this has been merged and will release with version 2.32.0 of the Terraform AWS Provider, later today. Thanks to @joelthompson for the implementation. 🎉
